x-pack/build_chromium/README.md

```diff
@@ -63,170 +50,72 @@ the Chromium repo to be cloned successfully. If checking out the Chromium fails
 
 ## Artifacts
 
 After the build completes, there will be a .zip file and a .md5 file in `~/chromium/chromium/src/out/headless`. These are named like so: `chromium-{first_7_of_SHA}-{platform}-{arch}`, for example: `chromium-4747cc2-linux-x64`.
+The zip files and md5 files are copied to a staging bucket in GCP storage.
 
-The zip files need to be deployed to GCP Storage. For testing, I drop them into `headless-shell-dev`, but for production, they need to be in `headless-shell`. And the `x-pack/plugins/reporting/server/browsers/chromium/paths.ts` file needs to be upated to have the correct `archiveChecksum`, `archiveFilename`, `binaryChecksum` and `baseUrl`. Below is a list of what the archive's are:
-
-- `archiveChecksum`: The contents of the `.md5` file, which is the `md5` checksum of the zip file.
-- `binaryChecksum`: The `md5` checksum of the `headless_shell` binary itself.
-
-*If you're building in the cloud, don't forget to turn off your VM after retrieving the build artifacts!*
-
-## Diagnosing runtime failures
-
-After getting the build to pass, the resulting binaries often failed to run or would hang.
-
-You can run the headless browser manually to see what errors it is generating (replace the `c:\dev\data` with the path to a dummy folder you've created on your system):
-
-**Mac**
-`headless_shell --disable-translate --disable-extensions --disable-background-networking --safebrowsing-disable-auto-update --disable-sync --metrics-recording-only --disable-default-apps --mute-audio --no-first-run --disable-gpu --no-sandbox --headless --hide-scrollbars --window-size=400,400 --remote-debugging-port=9221 https://example.com/`
-
-**Linux**
-`headless_shell --disable-translate --disable-extensions --disable-background-networking --safebrowsing-disable-auto-update --disable-sync --metrics-recording-only --disable-default-apps --mute-audio --no-first-run --disable-gpu --no-sandbox --headless --hide-scrollbars --window-size=400,400 --remote-debugging-port=9221 https://example.com/`
+## Testing
+Search the Puppeteer Github repo for known issues that could affect our use case, and make sure to test anywhere that is affected.
 
-**Windows**
-`headless_shell.exe --disable-translate --disable-extensions --disable-background-networking --safebrowsing-disable-auto-update --disable-sync --metrics-recording-only --disable-default-apps --mute-audio --no-first-run --disable-gpu --no-sandbox --headless --hide-scrollbars --window-size=400,400 --remote-debugging-port=9221 https://example.com/`
+Here's the steps on how to test a Puppeteer upgrade, run these tests on Mac, Windows, Linux x64 and Linux arm64:
 
-In the case of Windows, you can use IE to open `http://localhost:9221` and see if the page loads. In mac/linux you can just curl the JSON endpoints: `curl http://localhost:9221/json/list`.
+- Make sure the Reporting plugin is fetching the correct version of the browser
```

Comment:
This is a very good question. I think the only way this could work on all platforms is to do something like this:
```
x-pack/plugins/reporting/chromium/<platform_path>/<binary path> --headless --repl
```

and then type: `navigator.appVersion` into the REPL.

Seems like this should be an easier one-liner, or something that is handled by our helper script: `src/dev-tools/chromium_version.ts`